service/src/main/java/org/apache/griffin/core/metastore/hive/HiveMetaStoreServiceJdbcImpl.java

```diff
@@ -0,0 +1,287 @@
+            StorageDescriptor sd = new StorageDescriptor();
+            sd.setLocation(location);
+            sd.setCols(cols);
+            result.setSd(sd);
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            closeConnection(conn, stmt, rs);
+        }
+        return result;
+    }
+
+    @Scheduled(fixedRateString =
+            "${cache.evict.hive.fixedRate.in.milliseconds}")
+    @CacheEvict(
+            cacheNames = "hivejdbc",
+            allEntries = true,
+            beforeInvocation = true)
+    public void evictHiveCache() {
+        LOGGER.info("Evict hive cache");
+    }
+
+    private void closeConnection(Connection conn, Statement stmt, ResultSet rs) {
+        try {
+            if (rs != null) {
+                rs.close();
+            }
+            if (stmt != null) {
+                stmt.close();
+            }
+            if (conn != null) {
+                conn.close();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Query Hive for Show tables or show databases, which will return List of String
+     * @param sql sql string
+     * @return
+     */
+    private Iterable<String> queryHiveString(String sql) {
+        List<String> res = new ArrayList<>();
+        Statement stmt = null;
+        ResultSet rs = null;
+
+        try {
+            Class.forName(hiveClassName);
+            if (conn == null) {
+                conn = DriverManager.getConnection(hiveUrl);
+            }
+            LOGGER.info("got connection");
+        } catch (ClassNotFoundException | SQLException e) {
+            e.printStackTrace();
+        }
+
+        try {
+            stmt = conn.createStatement();
+            rs = stmt.executeQuery(sql);
+            while (rs.next()) {
+                res.add(rs.getString(1));
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            closeConnection(conn, stmt, rs);
+        }
+        return res;
+    }
+
+    /**
+     * Get the Hive table location from hive table metadata string
+     * @param tableMetadata hive table metadata string
+     * @return Hive table location
+     */
+    public String getLocation(String tableMetadata) {
+        tableMetadata = tableMetadata.toLowerCase();
+        int index = tableMetadata.indexOf("location");
+        if (index == -1) {
+            return "";
+        }
+
+        int start = tableMetadata.indexOf("\'", index);
+        int end = tableMetadata.indexOf("\'", start + 1);
+
+        if (start == -1 || end == -1) {
+            return "";
+        }
+
+        return tableMetadata.substring(start + 1, end);
+    }
+
+    /**
+     * Get the Hive table schema: column name, column type, column comment
+     * @param tableMetadata hive table metadata string
+     * @return List of FieldSchema
+     */
+    public List<FieldSchema> getColums(String tableMetadata) {
```

Comment:
Can you give an example for tableMetadata in getColums() Javadoc?

Reply:
done